add '| bool' on many conditionals

DO1JLR opened the pull request:
By running ceph-ansible there are a lot ``[DEPRECATION WARNING]`` like these:
```
[DEPRECATION WARNING]: evaluating containerized_deployment as a bare variable,
this behaviour will go away and you might need to add |bool to the expression
in the future. Also see CONDITIONAL_BARE_VARS configuration toggle.. This
feature will be removed in version 2.12. Deprecation warnings can be disabled
by setting deprecation_warnings=False in ansible.cfg.
```

Now appended ``| bool`` on a lot of the affected variables.

Sometimes the coding style from ``variable|bool`` changed to ``variable | bool`` *(with spaces at the pipe)*.

I hope you like it. If I should change something or did something completly wrong... Please say it!

guits commented:
@DO1JLR Thanks for this PR.

Could you add a sign off in your commit please?

guits commented:
Could you add the `Closes: #4022` keyword in your commit message as well?

Thanks!

DO1JLR commented:
> Could you add a sign off in your commit please?
 I added ``Signed-off-by: L3D <l3d@c3woc.de>``. I hope that's alright!

> Could you add the Closes: #4022 keyword in your commit message as well?
I added it..

I hope that's all right or did I miss something?

guits commented:
@DO1JLR looks good to me, let's wait for the CI

Thanks

guits commented on roles/ceph-handler/handlers/main.yml:
```diff
@@ -48,10 +48,10 @@
         # We do not want to run these checks on initial deployment (`socket.rc == 0`)
-        - mon_group_name in group_names
-        - containerized_deployment
+        - mon_group_name in group_names | bool
```
```suggestion
        - mon_group_name in group_names
```

DO1JLR replied:
Upsi, you are right.

DO1JLR commented:
I rebased the pull request with the current master and did the change @guits requested.

dsavineau commented on roles/ceph-handler/handlers/main.yml:
```diff
@@ -48,10 +48,10 @@
         # We do not want to run these checks on initial deployment (`socket.rc == 0`)
-        - mon_group_name in group_names
-        - containerized_deployment
+        - mon_group_name in group_names 
```
trailling space

DO1JLR replied:
Sorry, I removed the space and force-pushed this change.

dsavineau commented on roles/ceph-handler/handlers/main.yml:
```diff
@@ -48,10 +48,10 @@
-        - hostvars[item]['_mon_handler_called'] | default(False)
+        - hostvars[item]['_mon_handler_called'] | default(False) | bool
         - ceph_mon_container_stat.get('stdout_lines', [])|length != 0
         - not rolling_update
```
missing filter

DO1JLR replied:
```
- not rolling_update | bool
```
| bool filter is added.

dsavineau commented on roles/ceph-infra/tasks/configure_firewall.yml:
```diff
@@ -8,7 +8,7 @@
+  when: not containerized_deployment | bool
 
 - when: (firewalld_pkg_query.get('rc', 1) == 0
       or is_atomic)
```
missing filter

DO1JLR replied:
```
- when: (firewalld_pkg_query.get('rc', 1) == 0
      or is_atomic | bool)
```
is_atomic has now a boolean filter too!

dsavineau commented on tests/functional/rgw_multisite.yml:
```diff
@@ -21,13 +21,13 @@
       when:
-        - rgw_zonemaster
+        - rgw_zonemaster | bool
         - containerized_deployment | default(False)
```
missing filter

DO1JLR replied:
```
        - containerized_deployment | default(False) | bool
```
How could I miss this?
Now the filter is there too and the change is force pushed and amended in this commit here.

DO1JLR commented:
> some missing bool filter.
> You can take a look to #3830 to compare both PR.

Thanks @dsavineau 
I added the missing filter in my commit.

But I do not really understand why this pull request might now be merged, but # 3830 was closed.

guits commented:
@DO1JLR #3830 was opened a bit too early and had to be rebased on master too often because of the number of changes it had obviously.

guits commented:
jenkins test pipeline

DO1JLR commented:
I can see jenkins failed with the [ceph-ansible-prs-dev-centos-container-dashboard](https://2.jenkins.ceph.com/job/ceph-ansible-pipeline/81/)  Test.
But I can't identify what exactly the error is. Sorry.

@guits, Maybe you can help or tell me what went wrong?

guits commented:
@DO1JLR sorry for the late reply, the failure in ceph-ansible-prs-dev-centos-container-dashboard was unrelated to your patch.

could you rebase + fix conflicting files? it should be good to go after that.
Thanks!

DO1JLR commented:
> @DO1JLR sorry for the late reply, the failure in ceph-ansible-prs-dev-centos-container-dashboard was unrelated to your patch.
> 
> could you rebase + fix conflicting files? it should be good to go after that.
> Thanks!

Thank YOU for your great work wirh ceph-ansible and your patience with me.

I fixed the conflicting files and rebased it.

DO1JLR commented:
I made a mistake in the ``roles/ceph-infra/tasks/configure_firewall.yml`` file and force-pushed the correction.

DO1JLR commented:
Thanks :heart:
